resource/index.go

```diff
@@ -83,21 +89,42 @@ func (r *index) GetResource(path string, parent *Resource) (*Resource, bool) {
 }
 
 // GetResources returns first level resources.
-func (r *index) GetResources() []*Resource {
-	return r.resources
+func (i *index) GetResources() []*Resource {
+	return i.resources
 }
 
-func compileResourceGraph(resources subResources) error {
-	for _, r := range resources {
-		if err := r.Compile(); err != nil {
-			sep := "."
-			if err.Error()[0] == ':' {
-				sep = ""
+// resourceLookup provides a wrapper for Index that implements the  schema.ReferenceChecker interface.
+type refChecker struct {
+	index Index
+}
+
+// ReferenceChecker implements the schema.ReferenceChecker interface.
+func (rc refChecker) ReferenceChecker(path string) schema.FieldValidator {
+	rsc, exists := rc.index.GetResource(path, nil)
+	if !exists {
+		return nil
+	}
+	validator := rsc.Schema().Fields["id"].Validator
+
+	return schema.FieldValidatorFunc(func(value interface{}) (interface{}, error) {
+		var id interface{}
+		var err error
+
+		if validator != nil {
+			id, err = validator.Validate(value)
+			if err != nil {
+				return nil, err
 			}
-			return fmt.Errorf("%s%s%s", r.name, sep, err)
+		} else {
+			id = value
 		}
-	}
-	return nil
+
+		_, err = rsc.Get(context.TODO(), id)
```

Comment:
From the docs on `context.TODO()`:

> Code should use context.TODO when it's unclear which Context to use or it is not yet available (because the surrounding function has not yet been extended to accept a Context parameter).

From the docs on `context.Background()`:

> It is typically used by the main function, initialization, and tests, and as the top-level Context for incoming requests.

FroI feel this case is closer to the description offered for `context.TODO()` than what's offered by `context.Background()`. I think if we where to use the _right_ context here, we should get it passed in from somewhere so that when using the `rest` package, we use the context created for a new request, and not a new context created later.